### DISCUSSION TITLE
Firefox relay not working

### DISCUSSION THREAD

**Discussion by omersht** [dani-garcia/vaultwarden#2798] [Ideas]
currently from the supported email mask service firefox relay is the best free option,
But while firefox relay is showing as a thing in the ui every time i am trying to use it i get a error while this is working in the official bitwarden cloud, does someone manage to get it to work?

**Comment by stefan0xC** [comment]
The CSP for this service has been added (in 006a2aacbbb31736edb67b5887842813728daf38) but later removed (in 94be67eac11bf893dfc87d65a6e172cfcf3fdf37) because the firefox relay service itself apparently [only authorizes](https://github.com/mozilla/fx-private-relay/blob/7762ff630b79869ccf246f5a7a6017da30d24daf/privaterelay/settings.py#L970) the official bitwarden SaaS.

**Reply by BlackDex** [reply]
Indeed, not even the Self-Hosted Bitwarden supports this.

**Reply by dwarf-king-hreidmar** [reply]
Why does this continue to work from my browser plugin even though my plugin is using my self hosted vaultwarden server?

**Reply by BlackDex** [reply]
Because it's client side handled, not server side. The communication is directly with the API endpoint of the service, not your self-hosted instances
